From ad348ba6aed59be4027fa9178a46f35f8b6d8624 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 3 Dec 2015 19:01:45 +0200
Subject: [PATCH 0580/1240] spl: a3700: Initialize UART before call to early
 SPL init

- Allow debugging of early SPL init by initializing UART
  prior to early_spl_init() function
- UART1 and UART2 clocks are sourced from XTAL by default
  (see RD0012010 register for details). Additionally the GPIO
  control (RD0013830) sets the GPIO1[26:25] as UART1 pins by
  default.
  Therefore it is safe to start using UART before clocks
  and GPIO configuration.

Change-Id: I7d90d3b83dcff8eb0d3ba7b83c682498ebe54d24
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/25606
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm/cpu/mvebu-common/spl.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/arch/arm/cpu/mvebu-common/spl.c b/arch/arm/cpu/mvebu-common/spl.c
index 332a9d1..ac60ed9 100644
--- a/arch/arm/cpu/mvebu-common/spl.c
+++ b/arch/arm/cpu/mvebu-common/spl.c
@@ -51,12 +51,16 @@ void board_init_f(ulong silent)
 	/* Update the pointer to the FDT */
 	setup_fdt();
 
-	/* Init all drivers before UART initialization */
-	early_spl_init();
-
-	/* UART init */
+	/* UART1 and UART2 clocks are sourced from XTAL by default
+	* (see RD0012010 register for the details). Additionally the GPIO
+	* control (RD0013830) sets the GPIO1[26:25] as the UART1 pins by default.
+	* Therefore it is safe to start using UART before call to early_spl_init()
+	*/
 	preloader_console_init();
 
+	/* Init all drivers requred at early stage (clocks, GPIO...) */
+	early_spl_init();
+
 #ifdef CONFIG_MVEBU_SPL_SAR_DUMP
 	/* Sample at reset dump register */
 	mvebu_sar_dump_reg();
-- 
1.9.1

